packages/react/src/components/Table/EmptyTable/EmptyTable.jsx

```diff
@@ -86,7 +86,7 @@ const EmptyTable = ({
                 />
               ) : (
                 <EmptyState
-                  icon={emptyStateIcon ?? 'empty'}
+                  icon={emptyStateIcon !== '' ? emptyStateIcon : 'empty'}
```

Comment:
I see this is reverting, but what was the original code before you did any of this enumeration stuff.

The default value for emptyStateIcon is '', so in the previous change if emptyStateIcon is undefined, then it would bet set to '' and therefor no icon.  In the new change if emptyStateIcon is undefined it will be set to '' and based on the condition it will use 'empty'.  Is that the correct behaviour?

was the original behaviour such that if the prop was no specified then it would use the 'empy' icon?  if so, then I think we are good.

Reply:
The empty fallback would work only if `emptyStateIcon` is **undefined or null** in the `defaultProps` since this is an empty string, that is a "valid value" for the Nullish coalescing operator...
So if we want it to work with `??` the default props need to be `undefined/null` otherwise makes sense to revert to this code

Reply:
@cgirani The challenge (and what I'm trying to get to, is the default behaviour before the changes were in place).

This "revert" comes as a request from a downstream consumer where this change caused changes in the default behaviour of the component.  (I suspect that previously they were not setting the emptyStateIcon and was seeing empty, and after the bullish coalescing operator was used they got no icon (because the default value is an empty string it can never be undefined).

as a rule of thumb when we update components adding new features, etc, we never want to change the default behaviour of the component and I think this change did that.